packages/dom/src/utils.js

```diff
@@ -8,7 +8,7 @@ export function resourceFromDataURL(uid, dataURL) {
   // build a URL for the serialized asset
   let [, ext] = mimetype.split('/');
   let path = `/__serialized__/${uid}.${ext}`;
-  let url = new URL(path, document.URL).toString();
+  let url = document.URL.includes('localhost') ? new URL(path, 'http://render.percy.local').toString() : new URL(path, document.URL).toString();
```

Comment:
This should handle test cases mentioned below. 

```suggestion
  let url = url.replace(/(http[s]{0,1}:\/\/)localhost[:\d+]*/, '$1render.percy.local');
```